src/DotNetEnv/Parsers.cs

```diff
@@ -173,17 +173,23 @@ private static string ToUtf32Char (string hex)
                 .Or(OctalChar)
                 .Or(EscapedChar);
 
+        internal static readonly Parser<IEnumerable<char>> InlineCommentBeginOrControl =
+            Parse.Regex("[ \t]#").Or(Parse.Char(c => char.IsControl(c) && c != '\t', "controls except tab").Once());
```

Comment:
So, one of the goals I had with using the `Parser` approach was actually to avoid `Regex`. I am not 100% certain it can be avoided here, as it has been a very long time since I really thought about this code, so understanding your changes is a bit tricky for me at the moment, but my gut reaction is that I don't want to introduce regexes for handling comments differently than they are now, ideally.

Reply:
Looking closer at the changes, I see that I already had the regex and you were just carrying it over, so that makes total sense. I still would love a solution that was cleaner, ideally, but my bigger concern at this point is the seemingly new behavior below, of silently discarding invalid lines.

Reply:
I think you can do something like `Parse.String(" #").Or(Parse.String("\t#")).Or(...)`. (at least the tests still succed, so it should be correct I guess)
In my opinion such a short Regex is cleaner, but that's personal preference too.

About other concerns I will comment at the appropriate comments ;)